tests/datasets/test_dota.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+import torch
+import torch.nn as nn
+from _pytest.fixtures import SubRequest
+from pytest import MonkeyPatch
+
+from torchgeo.datasets import DOTA, DatasetNotFoundError
+
+pytest.importorskip('pyarrow')
+
+
+class TestDOTA:
+    @pytest.fixture(params=product(['train', 'val'], ['1.0', '2.0']))
+    def dataset(
+        self, monkeypatch: MonkeyPatch, tmp_path: Path, request: SubRequest
+    ) -> DOTA:
+        url = os.path.join('tests', 'data', 'dota', '{}')
+        monkeypatch.setattr(DOTA, 'url', url)
+
+        file_info = {
+            'train': {
+                'images': {
+                    '1.0': {
+                        'filename': 'dotav1_images_train.tar.gz',
+                        'md5': '14296c11c897cb7718558815a2b1bf69',
+                    },
+                    '2.0': {
+                        'filename': 'dotav2_images_train.tar.gz',
+                        'md5': 'fc80227b1f9b99cf5a7a3d0c5798efd0',
+                    },
+                },
+                'annotations': {
+                    '1.0': {
+                        'filename': 'dotav1_annotations_train.tar.gz',
+                        'md5': '805dc01688e00895a594c637569a2e1a',
+                    },
+                    '2.0': {
+                        'filename': 'dotav2_annotations_train.tar.gz',
+                        'md5': '723bceb26bc52a5de45902fada335c36',
+                    },
+                },
+            },
+            'val': {
+                'images': {
+                    '1.0': {
+                        'filename': 'dotav1_images_val.tar.gz',
+                        'md5': 'a95acf48281b7fc800666974730aeffd',
+                    },
+                    '2.0': {
+                        'filename': 'dotav2_images_val.tar.gz',
+                        'md5': '7c4ebb3317f970b26de273cd7313d46f',
+                    },
+                },
+                'annotations': {
+                    '1.0': {
+                        'filename': 'dotav1_annotations_val.tar.gz',
+                        'md5': '435a4a77c62eff955dd30a1b2a13894f',
+                    },
+                    '2.0': {
+                        'filename': 'dotav2_annotations_val.tar.gz',
+                        'md5': '86b629c6c8a1d924841d34de4eeb87ec',
+                    },
+                },
+            },
+        }
+        monkeypatch.setattr(DOTA, 'file_info', file_info)
+
+        root = tmp_path
+        split, version = request.param
+        if version == '2.0':
+            bbox_orientation = 'obb'
+        else:
+            bbox_orientation = 'hbb'
+
+        transforms = nn.Identity()
+
+        return DOTA(
+            root,
+            split,
+            version=version,
+            bbox_orientation=bbox_orientation,
+            transforms=transforms,
+            download=True,
+            checksum=True,
+        )
+
+    def test_getitem(self, dataset: DOTA) -> None:
+        for i in range(len(dataset)):
```

Comment:
I don't usually iterate over the dataset, is this necessary?

Reply:
Yes, because need to check the cases for empty and existing bounding boxes